src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -501,6 +563,24 @@ public long getEndpointDowntime(InetAddressAndPort ep)
             return 0L;
     }
 
+    /**
+     * @return a list of the most recent gossip state transitions for all peers
+     */
+    public List<GossipStateTransition> recentGossipStateTransitions()
+    {
+        // flatten the gossipStateTransitions map into a list sorted by the timestamp
+        return gossipStateTransitionMap.values().stream()
```

Comment:
Thinking loud. As you mentioned this map size can be up to tens of megabytes, preparing a List from this can further pressure the memory with more megabytes. So instead of creating a list here, if we just return reference to the map and let the caller directly read the map and fill the vtable, can reduce memory footprint. What do you think?